src/shared/tpm2-util.c

```diff
@@ -1709,87 +1762,36 @@ static int find_signature(
                 Tpm2Context *c,
-                const Tpm2Handle *primary,
-                const Tpm2Handle *parent_session,
-                TPM2_SE session_type,
+                const Tpm2Handle *session,
                 uint32_t hash_pcr_mask,
-                uint16_t pcr_bank, /* If UINT16_MAX, pick best bank automatically, otherwise specify bank explicitly. */
+                uint16_t pcr_bank,
                 const void *pubkey,
                 size_t pubkey_size,
                 uint32_t pubkey_pcr_mask,
                 JsonVariant *signature_json,
                 bool use_pin,
-                Tpm2Handle **ret_session,
-                TPM2B_DIGEST **ret_policy_digest,
-                TPMI_ALG_HASH *ret_pcr_bank) {
+                TPM2B_DIGEST **ret_policy_digest) {
 
-        static const TPMT_SYM_DEF symmetric = {
-                .algorithm = TPM2_ALG_AES,
-                .keyBits.aes = 128,
-                .mode.aes = TPM2_ALG_CFB,
-        };
         TSS2_RC rc;
         int r;
 
         assert(c);
+        assert(session);
         assert(pubkey || pubkey_size == 0);
         assert(pubkey_pcr_mask == 0 || pubkey_size > 0);
 
-        log_debug("Starting authentication session.");
-
-        /* So apparently some TPM implementations don't implement trial mode correctly. To avoid issues let's
-         * avoid it when it is easy to. At the moment we only really need trial mode for the signed PCR
-         * policies (since only then we need to shove PCR values into the policy that don't match current
-         * state anyway), hence if we have none of those we don't need to bother. Hence, let's patch in
-         * TPM2_SE_POLICY even if trial mode is requested unless a pubkey PCR mask is specified that is
-         * non-zero, i.e. signed PCR policy is requested.
-         *
-         * One day we should switch to calculating policy hashes client side when trial mode is requested, to
-         * avoid this mess. */
-        if (session_type == TPM2_SE_TRIAL && pubkey_pcr_mask == 0)
-                session_type = TPM2_SE_POLICY;
+        log_debug("Building sealing policy.");
 
         if ((hash_pcr_mask | pubkey_pcr_mask) != 0) {
-                /* We are told to configure a PCR policy of some form, let's determine/validate the PCR bank to use. */
-
-                if (pcr_bank != UINT16_MAX) {
-                        r = tpm2_pcr_mask_good(c, pcr_bank, hash_pcr_mask|pubkey_pcr_mask);
-                        if (r < 0)
-                                return r;
-                        if (r == 0)
-                                log_warning("Selected TPM2 PCRs are not initialized on this system, most likely due to a firmware issue. PCR policy is effectively not enforced. Proceeding anyway.");
-                } else {
-                        /* No bank configured, pick automatically. Some TPM2 devices only can do SHA1. If we
-                         * detect that use that, but preferably use SHA256 */
```

Comment:
this comment was lost when moving

Reply:
ack, I partially added it back in; I left out the part about 'no bank configured' since the code is now in tpm2_seal() which doesn't provide a way to specify the bank, so it always picks the best bank.